x/authz/keeper/keeper.go

```diff
@@ -191,6 +190,19 @@ func (k Keeper) DispatchActions(ctx context.Context, grantee sdk.AccAddress, msg
 func (k Keeper) SaveGrant(ctx context.Context, grantee, granter sdk.AccAddress, authorization authz.Authorization, expiration *time.Time) error {
 	sdkCtx := sdk.UnwrapSDKContext(ctx)
 	msgType := authorization.MsgTypeURL()
+
+	// Ensure grantee is a valid bech32 address
+	_, err := sdk.AccAddressFromBech32(grantee.String())
+	if err != nil {
+		return fmt.Errorf("failed to convert grantee address to bytes: %w", err)
+	}
+
+	// Ensure granter is a valid bech32 address
+	_, err = sdk.AccAddressFromBech32(granter.String())
+	if err != nil {
+		return fmt.Errorf("failed to convert granter address to bytes: %w", err)
+	}
```

Comment:
I'm aware that there's some other context on this PR, but I can tell you that this code doesn't make sense here. `String` does bech32 formatting and `AccAddressFromBech32` does bech32 parsing. You are just taking a byte slice, formatting it as bech32 and then parsing that back from bech32. It would be like calling `JSON.parse(JSON.stringify(...))`. This just adds extra overhead to the function call. Any `AccAddress`s should be pre-validated.